engines/scumm/he/sound_he.cpp

```diff
@@ -472,598 +816,877 @@ void SoundHE::processSoundOpcodes(int sound, byte *codePtr, int *soundVars) {
+		// Birthday Surprise to change the note on the piano
+		// when not using a digitized instrument.
+		_vm->_imuse->stopSound(_currentMusic);
+		_currentMusic = soundId;
+		_vm->_imuse->startSoundWithNoteOffset(soundId, heOffset);
+	} else if (_vm->_musicEngine) {
+		_vm->_musicEngine->stopSound(_currentMusic);
+		_currentMusic = soundId;
+		_vm->_musicEngine->startSoundWithTrackID(soundId, heOffset);
+	}
 }
 
-void SoundHE::startHETalkSound(uint32 offset) {
+void SoundHE::playVoice(uint32 offset, uint32 length) {
 	byte *ptr;
-	int32 size;
+	int talkieChannel = (_vm->VAR_TALK_CHANNEL != 0xFF) ? _vm->VAR(_vm->VAR_TALK_CHANNEL) : 0;
 
 	if (ConfMan.getBool("speech_mute"))
 		return;
 
 	if (_sfxFilename.empty()) {
 		// This happens in the Pajama Sam's Lost & Found demo, on the
 		// main menu screen, so don't make it a fatal error.
-		warning("startHETalkSound: Speech file is not found");
+		warning("SoundHE::playVoice(): Speech file is not found");
 		return;
 	}
 
 	ScummFile file(_vm);
 	if (!_vm->openFile(file, _sfxFilename)) {
-		warning("startHETalkSound: Could not open speech file %s", _sfxFilename.c_str());
+		warning("SoundHE::playVoice(): Could not open speech file %s", _sfxFilename.c_str());
 		return;
 	}
+
 	file.setEnc(_sfxFileEncByte);
 
 	// Speech audio doesn't have a unique ID,
 	// so we use the file offset instead.
 	// _heTalkOffset is used at tryLoadSoundOverride.
 	_heTalkOffset = offset;
 
-	_sfxMode |= 2;
-	_vm->_res->nukeResource(rtSound, 1);
+	_digiSndMode |= DIGI_SND_MODE_TALKIE;
+	_heMixer->stopChannel(talkieChannel);
+	_vm->_res->nukeResource(rtSound, HSND_TALKIE_SLOT);
 
-	file.seek(offset + 4, SEEK_SET);
-	size = file.readUint32BE();
 	file.seek(offset, SEEK_SET);
 
-	_vm->_res->createResource(rtSound, 1, size);
+	if (_vm->hasLocalizer()) {
+		file.seek(offset + 4, SEEK_SET);
+		length = file.readUint32BE();
+		file.seek(offset, SEEK_SET);
+	}
+
+	_vm->_res->createResource(rtSound, 1, length);
 	ptr = _vm->getResourceAddress(rtSound, 1);
-	file.read(ptr, size);
+	file.read(ptr, length);
+
+	addSoundToQueue(HSND_TALKIE_SLOT, 0, talkieChannel, 0, HSND_BASE_FREQ_FACTOR, HSND_SOUND_PAN_CENTER, HSND_MAX_VOLUME);
+}
 
-	int channel = (_vm->VAR_TALK_CHANNEL != 0xFF) ? _vm->VAR(_vm->VAR_TALK_CHANNEL) : 0;
-	addSoundToQueue2(1, 0, channel, 0);
+void SoundHE::playVoiceFile(char *filename) {
+	// This is unimplemented on purpose! I haven't found a game which uses this;
+	// after all, this is a development path and it doesn't seem to affect any game
+	// version I had the chance to test. Emphasis on "had the chance to test":
+	// I don't know if there's actually a game which uses this!
+	//
+	// Originally this served the purpose of fetching voice lines from separate files
+	// instead of using a single .HE2 bundle.
+	//
+	// Again, we should never end up in here, but IF WE DO we issue an error.
+	error("SoundHE::playVoiceFile(): Unimplemented development codepath, please file a ticket!");
```

Comment:
"... at https://bugs.scummvm.org". And maybe show MessageDialog with the URL to open it right away, so the chances that we get this reported are higher.